llmfoundry/models/layers/attention.py

```diff
@@ -581,6 +589,40 @@ def forward(
             query = self.q_ln(query).to(dtype)
             key = self.k_ln(key).to(dtype)
 
+        if rotary_emb_w_meta_info is not None:
+            rotary_emb = rotary_emb_w_meta_info['rotary_emb']
+            seq_len = rotary_emb_w_meta_info['seq_len']
+            offset_info = rotary_emb_w_meta_info['offset_info']
+
+            query = query.view(*(query.shape[:-1]), -1, self.head_dim)
```

Comment:
instead of these * unpacking for shapes, can you define variables for each dimension you need and use them directly?